.github/workflows/build-test.yaml

```diff
@@ -28,8 +28,8 @@ jobs:
       env:
           PYOPENGL_PLATFORM: osmesa
       run: pytest --html=report.html --self-contained-html
-    - uses: actions/upload-artifact@v3
+    - uses: actions/upload-artifact@v4
       name: Upload pytest report
       with:
-        name: pytest-report
+        name: pytest-report-py-${{ matrix.python-version }}
```

Comment:
v4 of upload-artifact will now error when uploading artifacts with the same name, so distinguishing by python version (could instead only upload once)